### ISSUE TITLE
Training Loop and Loss Functions

### ISSUE THREAD

**Issue by mahmoodkashmiri** [hamidriasat/PIDNet#1]
Thanks for the great work!

Any plans to release code for training loop and loss functions?

**Comment by hamidriasat** [comment]
You're welcome!
This can be used with any segmentation training pipeline.
[Here](https://keras.io/examples/vision/basnet_segmentation/) you can find one of my used segmentation training pipeline.
If you are looking for an advanced pipeline with more features, you can also replace this model with my contributed code [here](https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow2/Segmentation/Contrib/UNet3P).

**Comment by mahmoodkashmiri** [comment]
Awesome. Thank you very much Hamid.
